File: program/src/state/promisee.rs
use anchor_lang::prelude::*;

#[account]
pub struct Promisee {
    // Bump seed
    pub bump: u8,
    // The owner of this account
    pub owner: Pubkey,
    // The promise network this account belongs to
    pub promise_network: Pubkey,
    // The time this account was created
    pub created_at: i64,
    // The last time this account was updated
    pub updated_at: i64,
}

impl Promisee {
  pub const SEED_PREFIX: &'static [u8] = b"promisee";
  pub const SIZE: usize = 
    8 +
    8 + // bump
    32 + // owner
    32 + // promise_network
    8 + // created_at
    8; // updated_at
}

File: sdk/src/promisee/Promisee.ts
import { PublicKey } from "@solana/web3.js";

export type Promisee = {
  /** Public address of the Promisee */
  address: PublicKey;

  /** Bump of the Promisee */
  bump: Number;

  /** Public address of the owner of the Promisee */
  owner: PublicKey;

  /** Public address of the creator of the Promisee */
  creator: PublicKey;

  /** Public address of the Promise */
  promise: PublicKey;

  /** Date the Promisee was created */
  createdAt: Date;

  /** Date the Promisee was updated */
  updatedAt: Date;
};

export const createPromiseeSeeds = (
  promise: PublicKey,
  owner: PublicKey
): Buffer[] => {
  return [Buffer.from("promisee"), promise.toBuffer(), owner.toBuffer()];
};
